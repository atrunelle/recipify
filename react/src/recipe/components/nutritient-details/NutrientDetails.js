import React, { Component } from 'react';
import { Typography } from '@material-ui/core';

class NutrientDetails extends Component {
  render() {
    if (!this.props.nutrient.totalQuantity.quantity) return '';
    const totalQuantityQuantity = Math.round(this.props.nutrient.totalQuantity.quantity);
    const totalDailyQuantity = Math.round(this.props.nutrient.totalDaily.quantity);

    return (
      <div>
          <Typography
            gutterBottom
            component="h3">
            { this.props.nutrient.label } 
          </Typography>
          <Typography
            gutterBottom
            paragraph
            component="p">
            { totalQuantityQuantity } { this.props.nutrient.totalQuantity.unit }
          </Typography>
          <Typography
            gutterBottom
            paragraph
            component="p">
            { totalDailyQuantity } { this.props.nutrient.totalDaily.unit }
          </Typography>
      </div>
    );
  }
}

export default  NutrientDetails;