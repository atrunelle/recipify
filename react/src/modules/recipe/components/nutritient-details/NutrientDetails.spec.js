import React from 'react';
import { shallow } from 'enzyme';
import NutrientDetails from './NutrientDetails';

describe('Component: NutrientDetails', () => {
  it('should not render is no nutrient is passed as prop', () => {
    const component = shallow(<NutrientDetails />);
    expect(component.html()).toBeNull();
  });

  it('should match snapshot', () => {
    const mockNutrient = {
      totalQuantity: {
        label: 'Fat',
        quantity: 100,
        unit: 'gr',
      },
      totalDaily: {
        label: 'Fat',
        quantity: 10,
        unit: '%',
      },
    };

    const component = shallow(<NutrientDetails nutrient={mockNutrient} />);
    expect(component.html()).toMatchSnapshot();
  });
});