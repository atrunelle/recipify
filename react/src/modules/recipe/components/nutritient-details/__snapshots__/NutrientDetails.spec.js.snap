// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NutrientDetails should match snapshot 1`] = `"<div><h3 class=\\"MuiTypography-root-1 MuiTypography-body1-10 MuiTypography-gutterBottom-18\\"></h3><p class=\\"MuiTypography-root-1 MuiTypography-body1-10 MuiTypography-gutterBottom-18 MuiTypography-paragraph-19\\">100 gr</p><p class=\\"MuiTypography-root-1 MuiTypography-body1-10 MuiTypography-gutterBottom-18 MuiTypography-paragraph-19\\">10 %</p></div>"`;
